### PR TITLE
refactor(l2): l1_watcher make deposits unique

### LINKED ISSUE #1191: fix(L2): add uniqueness to deposit logs on L1
Currently when someone makes a deposit the result of `keccak(to || value)` is saved in the `depositLogs` list of the L1 bridge contract. As we will use this to check repeated transacitions, they should be unique.

Note: the nonce isn't a great choice for uniqueness since the deposit beneficiary can be different than the sender

### LINKED ISSUE #1279: L2: improve and refactor `l1_watcher`
The `l1_watcher` requires some changes. 
When the `proposer` is restarted, the `l1_watcher` processes all the events, resulting in duplicated `deposit` transactions. 
This behavior was observed when depositing the same amount twice. It may be related just to Issue #1191, or perhaps additional logic is required.
 
TaskList:
- [x] #1191

### PR DESCRIPTION
**Pull request by fborello-lambda**
**Motivation**

We need a way to identify deposit. In order to do it, we can use a "depositId" that works as a counter. It is being incremented every time a deposit event is emitted.

We've been starting the `l1_watcher` from block 0. Fetching logs from block 0 will throw errors if asking light_nodes and it's wasteful. We will need a more robust way to keep track of it, for now, we should start from the first block in which the `CommonBridge` was deployed.

**Description**

- [CommonBridge] Use the `depositId` to construct the deposit_hash (`keccak(to || value || depositId)`)
- [L1Watcher] Use the `depositId` from the logs to avoid duplicating the mint transaction. And use it as the mint transaction's `nonce`. In this way, the [Committer] can create the `StateDiff` to send the `commit` transaction to the OnChainProposer.
- Add useful Makefile targets.
- Add simple command to fetch the `depositId`.
- Start from the block in which the `CommonBridge` was deployed.

Closes #1191

### CODE REVIEW

File: crates/l2/.env.example [hunk 1]
```diff
@@ -11,7 +11,6 @@ DEPLOYER_PRIVATE_KEY=0x385c546456b6a603a1cfcaa9ec9494ba4832da08dd6bcf4de9a71e4a0
 # If set to false, the salt will be randomized.
 DEPLOYER_SALT_IS_ZERO=true
 L1_WATCHER_BRIDGE_ADDRESS=0x266ffef34e21a7c4ce2e0e42dc780c2c273ca440
-L1_WATCHER_TOPICS=0x6f65d68a35457dd88c1f8641be5da191aa122bc76de22ab0789dcc71929d7d37
```
Comment (jrchatruc) [inline crates/l2/.env.example:14]: Why is this line deleted?

Comment (fborello-lambda) [reply crates/l2/.env.example:14]: Because if we change the event we will have to recalculate the hash again, i've replaced the envar with some code that does this for us:
```rust
let topic = keccak(b"DepositInitiated(uint256,address,uint256,bytes32)");
```

**Review by jrchatruc — APPROVED** [review]
